deepchem/feat/mol_graphs.py

```diff
@@ -264,8 +264,11 @@ def agglomerate_mols(mols, max_deg=10, min_deg=0):
     num_mols = len(mols)
 
     # Combine the features, then sort them by (atom_degree, mol_index)
-    atoms_by_deg = np.concatenate([x.atom_features for x in mols])
-    degree_vector = np.concatenate([x.degree_list for x in mols], axis=0)
+    try:
+      atoms_by_deg = np.concatenate([x.atom_features for x in mols])
+      degree_vector = np.concatenate([x.degree_list for x in mols], axis=0)
+    except:
+      raise RuntimeError('Incorrect featurization used for Convolution model')
```

Comment:
I like the try/catch statement approach, but we need to be a bit careful here. We don't want to catch all exceptions so we should make sure to catch the precise exception that is raised if ECFP is used with GraphConv (or another featurization)

Reply:
That's what my initial thought was as well but obtaining the dataset object and getting the model object are separate and the only time they come together is when the ```model.fit``` function is called.
This was the place where I found that the behaviours diverged depending on featurization type.

Reply:
One alternative might be to check that the entries in the incoming dataset are `ConvMol` objects (perhaps just checking the first object for speed) and raise an error if so

Reply:
Yeah I think that would be much faster but again that would happen in the `model.fit` function if I understand correctly.